internal/application/utils/utils.go

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package utils
+
+import "github.com/asgardeo/thunder/internal/application/model"
+
+func IsAllowedGrantType(oauthApplication *model.OAuthApplication, grantType string) bool {
```

Comment:
WDYT of moving the OAuth stuff to a separate go module, so that the thunder will be a monorepo of go modules? That way, it would be more organized, IMO.

Reply:
We'll consider this comment in future revamping